update-db.js

```diff
@@ -117,66 +100,100 @@ function diffBrowsersLists (old, current) {
     .join('\n')
 }
 
-function updateLockfile (lock, latest) {
-  if (lock.mode === 'npm') {
-    var fixed = deletePackage(JSON.parse(lock.content))
-    return JSON.stringify(fixed, null, '  ')
-  } else {
-    var lines = lock.content.split('\n')
-    var i
-    /* istanbul ignore if */
-    if (lock.mode === 'pnpm') {
-      for (i = 0; i < lines.length; i++) {
-        if (lines[i].indexOf('caniuse-lite:') >= 0) {
-          lines[i] = lines[i].replace(/: .*$/, ': ' + latest.version)
-        } else if (lines[i].indexOf('/caniuse-lite') >= 0) {
-          lines[i] = lines[i].replace(/\/[^/:]+:/, '/' + latest.version + ':')
-          for (i = i + 1; i < lines.length; i++) {
-            if (lines[i].indexOf('integrity: ') !== -1) {
-              lines[i] = lines[i].replace(
-                /integrity: .+/, 'integrity: ' + latest.dist.integrity
-              )
-            } else if (lines[i].indexOf(' /') !== -1) {
-              break
-            }
-          }
-        }
-      }
-    } else if (lock.mode === 'yarn') {
-      for (i = 0; i < lines.length; i++) {
-        if (lines[i].indexOf('caniuse-lite@') !== -1) {
-          lines[i + 1] = lines[i + 1].replace(
-            /version "[^"]+"/, 'version "' + latest.version + '"'
-          )
-          lines[i + 2] = lines[i + 2].replace(
-            /resolved "[^"]+"/, 'resolved "' + latest.dist.tarball + '"'
-          )
-          lines[i + 3] = lines[i + 3].replace(
-            /integrity .+/, 'integrity ' + latest.dist.integrity
-          )
-          i += 4
-        }
-      }
-    }
-    return lines.join('\n')
-  }
+function updateNpmLockfile (lock, latest) {
+  var metadata = { latest: latest, versions: [] }
+  var content = deletePackage(JSON.parse(lock.content), metadata)
+  metadata.content = JSON.stringify(content, null, '  ')
+  return metadata
 }
 
-function deletePackage (node) {
+function deletePackage (node, metadata) {
   if (node.dependencies) {
-    delete node.dependencies['caniuse-lite']
+    if (node.dependencies['caniuse-lite']) {
+      var version = node.dependencies['caniuse-lite'].version
+      metadata.versions[version] = true
+      delete node.dependencies['caniuse-lite']
+    }
     for (var i in node.dependencies) {
-      node.dependencies[i] = deletePackage(node.dependencies[i])
+      node.dependencies[i] = deletePackage(node.dependencies[i], metadata)
     }
   }
   return node
 }
 
-module.exports = function updateDB (print) {
-  var lock = detectLockfile()
+var yarnVersionRe = new RegExp('version "(.*?)"')
+
+function updateYarnLockfile (lock, latest) {
+  var blocks = lock.content.split(/(\n{2,})/).map(function (block) {
+    return block.split('\n')
+  })
```

Comment:
It's easier to parse the yarn lock file if you split it up into blocks separated by two or more `\n`s, which are then split into lines. Apart from the first block (header material), the blocks have the same format.